lib/spack/spack/test/spec_syntax.py

```diff
@@ -433,6 +440,214 @@ def test_duplicate_architecture_component(self):
         ]
         self._check_raises(DuplicateArchitectureError, duplicates)
 
+    @pytest.mark.usefixtures('config')
+    def test_parse_yaml_simple(self, mock_packages, tmpdir):
+        s = Spec('libdwarf')
+        s.concretize()
+
+        specfile = tmpdir.join('libdwarf.yaml')
+
+        with specfile.open('w') as f:
+            f.write(s.to_yaml(hash=ht.build_hash))
+
+        # Check an absolute path to spec.yaml by itself:
+        #     "spack spec /path/to/libdwarf.yaml"
+        specs = sp.parse(specfile.strpath)
+        assert len(specs) == 1
+
+        # Check absolute path to spec.yaml mixed with a clispec, e.g.:
+        #     "spack spec mvapich_foo /path/to/libdwarf.yaml"
+        specs = sp.parse('mvapich_foo {0}'.format(specfile.strpath))
+        assert len(specs) == 2
+
+    @pytest.mark.usefixtures('config')
+    def test_parse_namespaced_yaml_package(self, mock_packages, tmpdir):
```

Comment:
@tgamblin I think a clearer name for this would be `test_parse_filename_missing_slash` (this being misinterpreted as a weirdly-namespaced yaml package is a symptom).

With this test I think you can also remove the corresponding check from `test_parse_yaml_relative_paths`:

```
            # Test for command like: "spack spec libelf.yaml"
            specs = sp.parse('{0}'.format(file_name))
```

I think that would be good because everything is explained here and I think in order to avoid confusion a similar explanation would have to be there too.

Everything else looks good but I think it would be worthwhile to make this change before merging.